### PR TITLE
Fix wrong memory measurements of containers and vms

### PR DESCRIPTION
**Pull request by tlusser-inv**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by tlusser-inv** [comment]
Linked PR to issue #8291

File: plugins/inputs/proxmox/proxmox.go [hunk 1]
```diff
@@ -137,7 +139,12 @@ func gatherVmData(px *Proxmox, acc telegraf.Accumulator, rt ResourceType) {
 			return
 		}
 		tags := getTags(px, vmStat.Name, vmConfig, rt)
-		fields, err := getFields(vmStat)
+		currentVMStatus, err := getCurrentVMStatus(px, rt, vmStat.ID)
```
Comment (ssoroka) [inline plugins/inputs/proxmox/proxmox.go:142]: Just a note that this adds an http request inside a loop. It's going to have some negative performance implications, especially on systems with a large number of VMs. If there's any way to get this data in aggregate with one call, that would be preferable.

**Review by ssoroka — CHANGES_REQUESTED** [review]
looks good. I'd like to avoid the NodeName blank check before merge. see comment.

File: plugins/inputs/proxmox/proxmox.go [hunk 2]
```diff
@@ -48,11 +48,13 @@ func (px *Proxmox) Gather(acc telegraf.Accumulator) error {
...
 func (px *Proxmox) Init() error {
-	hostname, err := os.Hostname()
-	if err != nil {
-		return err
+	if px.NodeName == "" {
+		hostname, err := os.Hostname()
+		if err != nil {
+			return err
+		}
+		px.NodeName = hostname
```
[the remark this answers is not in the document]
Comment (tlusser-inv) [reply plugins/inputs/proxmox/proxmox.go:56]: This change is to be backwards compatible (no config file change needed) and also allows testing against a proxmox instance which is not running on the current host.

File: plugins/inputs/proxmox/proxmox.go (hunk 1 above)
Comment (tlusser-inv) [reply plugins/inputs/proxmox/proxmox.go:142]: We are aware of this, but AFAIK there is no other possibility in the REST-API to query the status of all virtual machines containers in one step (please see: https://pve.proxmox.com/pve-docs/api-viewer/)

File: plugins/inputs/proxmox/proxmox.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tlusser-inv) [reply plugins/inputs/proxmox/proxmox.go]: Moved to private init function

**Review by ssoroka — APPROVED** [review]

**Comment by ssoroka** [comment]
Awesome, Thanks!

**Pull request merged**